MusicTaste/MusicTaste/MatchingAlgorithm.m

```diff
@@ -58,35 +54,36 @@ + (double) compareSpotifyData:(PFObject *)potentialMatch withData:(NSManagedObje
-                [matchDataArtists removeObjectsInArray:[potentialMatch valueForKey:@"artists"]];
+                [matchDataGenresCopy removeObjectsInArray:[potentialMatch valueForKey:@"genres"]];
+                [matchDataTracksCopy removeObjectsInArray:[potentialMatch valueForKey:@"tracks"]];
+                [matchDataAlbumsCopy removeObjectsInArray:[potentialMatch valueForKey:@"albums"]];
+                [matchDataArtistsCopy removeObjectsInArray:[potentialMatch valueForKey:@"artists"]];
                 //gets the new count of the array
-                double newGenreCount = [matchDataGenres count];
-                double newAlbumCount = [matchDataAlbums count];
-                double newTrackCount = [matchDataTracks count];
-                double newArtistCount = [matchDataArtists count];
+                double newGenreCount = [matchDataGenresCopy count];
+                double newAlbumCount = [matchDataAlbumsCopy count];
+                double newTrackCount = [matchDataTracksCopy count];
+                double newArtistCount = [matchDataArtistsCopy count];
                 //generates a integer to determine how similar the two user's music taste is - 0 for similar and 1 for not similar at all
-                spotifyMatch = (((newGenreCount/genreCount)*0.2) + ((newTrackCount/trackCount)*0.3) + ((newAlbumCount/albumCount)*0.2) + ((newArtistCount/artistCount)*0.3));
+                if (genreCount == 0 || albumCount == 0 || trackCount == 0 || albumCount == 0){
+                    //To avoid dividing by zero if any of the counts are zero we make the Spotify match automatically 1
+                    self.spotifyMatch = 1;
+                }else{
+                self.spotifyMatch = (((newGenreCount/genreCount)*0.2) + ((newTrackCount/trackCount)*0.3) + ((newAlbumCount/albumCount)*0.2) + ((newArtistCount/artistCount)*0.3));
+                }
+                [matchesDict setObject:[NSNumber numberWithDouble:self.spotifyMatch] forKey:[matchData valueForKey:@"objectId"]];
+                [matches addObject:matchesDict];
+                }
             }
-        }];
-    });
-    return spotifyMatch;
-}
-
-+ (void) compare:(PFObject *)potentialMatch withDictionary:(NSMutableDictionary *)matches withData:(NSManagedObject *)data {
-    PFQuery *query = [PFQuery queryWithClassName:@"Music"];
-    [query findObjectsInBackgroundWithBlock:^(NSArray *results, NSError *error) {
-        //Creates a double that will be used to measure the similarity between users
-        double spotifyComp = 0.0;
-        if (results != NULL){
-            //Calls upon compareSpotifyData to compare the user to another user
-            spotifyComp = [self compareSpotifyData:potentialMatch withData:results];
-        }
-        [matches setObject:[NSNumber numberWithDouble:spotifyComp] forKey:potentialMatch.objectId];
-    }];
+    }
+    //Saves matches to Database
+    potentialMatch[@"matches"] = matches;
```

Comment:
Adds the matches Array to the current potential match's database

Reply:
If I'm reading the algorithm correctly, it seems that we're doing something like the following

```
get Users table from Parse (network call)
for each user in the Users table
    compare that user to every other user
    save the matches for that user to Parse (network call)
```

Do we really need to save all matches for everyone? Or can we save the matches just for the logged in user